hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go

```diff
@@ -75,7 +76,12 @@ func reconcileOpenStackClusterSpec(hcluster *hyperv1.HostedCluster, openStackClu
 	// If no MachineNetwork is provided, use a default CIDR block.
 	// Note: The default is required for now because there is no CLI option to set the MachineNetwork.
 	// See https://github.com/openshift/hypershift/pull/4287
+	// When subnet is provided, machine network is required.
+	// Note: currently there is no validation that the provided subnets are within the machine network, it's up to the user to ensure that for now.
 	if hcluster.Spec.Networking.MachineNetwork == nil || len(hcluster.Spec.Networking.MachineNetwork) == 0 {
+		if openStackPlatform.ManagedSubnets != nil && len(openStackPlatform.ManagedSubnets) > 0 {
+			return fmt.Errorf("failed to reconcile OpenStack CAPI cluster, machine network is required when subnets are provided")
```

Comment:
Is the `MachineNetwork` now required because we need the CIDR in the `getIngressIP`?
If that is the case, I don't think we need to add a default CIDR, in the next line, anymore right?

Reply:
this validation returns an error if MachineNetwork is not provided via CLI or YAML and if the user provides ManagedSubnets or Subnets are pre-created (I actually forgot a condition on the later, I'll add it now).
Does it make sense?